platform/core/src/utils/isDisplaySetReconstructable.js

```diff
@@ -31,37 +33,43 @@ export default function isDisplaySetReconstructable(instances) {
 
 function processMultiframe(instance) {
   //TODO: deal with multriframe checks! return false for now as can't reconstruct.
-  return { value: false };
+  const warningIssues = [reconstructionIssues.MULTIFRAMES];
+  return { value: false, warningIssues };
 }
 
 function processSingleframe(instances) {
+  const n = instances.length;
   const firstImage = instances[0].getData().metadata;
   const firstImageRows = firstImage.Rows;
   const firstImageColumns = firstImage.Columns;
   const firstImageSamplesPerPixel = firstImage.SamplesPerPixel;
   const firstImageOrientationPatient = firstImage.ImageOrientationPatient;
   const firstImagePositionPatient = firstImage.ImagePositionPatient;
 
+  const warningIssues = [];
   // Can't reconstruct if we:
   // -- Have a different dimensions within a displaySet.
   // -- Have a different number of components within a displaySet.
   // -- Have different orientations within a displaySet.
-  for (let i = 1; i < instances.length; i++) {
-    const instance = instances[i].getData().metadata;
+  for (let ii = 1; ii < n; ++ii) {
+    const instance = instances[ii].getData().metadata;
     const {
       Rows,
       Columns,
       SamplesPerPixel,
       ImageOrientationPatient,
     } = instance;
 
-    if (
-      Rows !== firstImageRows ||
-      Columns !== firstImageColumns ||
-      SamplesPerPixel !== firstImageSamplesPerPixel ||
-      !_isSameOrientation(ImageOrientationPatient, firstImageOrientationPatient)
-    ) {
-      return { value: false };
+    if (Rows !== firstImageRows || Columns !== firstImageColumns) {
+      warningIssues.push(reconstructionIssues.VARYING_IMAGESDIMENSIONS);
```

Comment:
Nice! Looks like reconstructionIssues is a enum. In js we usually use uppercase: ReconstructionIssues.VARYING_IMAGESDIMENSIONS.